x-pack/plugins/spaces/server/usage_stats/usage_stats_client.ts

```diff
@@ -0,0 +1,130 @@
+  Pick<ResolveConflictsOptions, 'createNewCopies'>;
+
+const COPY_DEFAULT = Object.freeze({
+  total: 0,
+  kibanaRequest: Object.freeze({ yes: 0, no: 0 }),
+  createNewCopiesEnabled: Object.freeze({ yes: 0, no: 0 }),
+  overwriteEnabled: Object.freeze({ yes: 0, no: 0 }),
+});
+const RESOLVE_COPY_ERRORS_DEFAULT = Object.freeze({
+  total: 0,
+  kibanaRequest: Object.freeze({ yes: 0, no: 0 }),
+  createNewCopiesEnabled: Object.freeze({ yes: 0, no: 0 }),
+});
+
+export class UsageStatsClient {
+  constructor(
+    private readonly debugLogger: (message: string) => void,
+    private readonly repository: ISavedObjectsRepository
+  ) {}
+
+  public async getUsageStats() {
+    this.debugLogger('getUsageStats() called');
+    let usageStats: UsageStats = {};
+    try {
+      const result = await this.repository.get<UsageStats>(
+        SPACES_USAGE_STATS_TYPE,
+        SPACES_USAGE_STATS_TYPE
+      );
+      usageStats = result.attributes;
+    } catch (err) {
+      // do nothing
+    }
+    return usageStats;
+  }
+
+  public async incrementCopySavedObjects({
+    headers,
+    createNewCopies,
+    overwrite,
+  }: IncrementCopySavedObjectsOptions) {
+    const usageStats = await this.getUsageStats();
+    const { apiCalls = {} } = usageStats;
+    const { copySavedObjects: current = COPY_DEFAULT } = apiCalls;
+
+    const attributes = {
+      ...usageStats,
+      apiCalls: {
+        ...apiCalls,
+        copySavedObjects: {
+          total: current.total + 1,
+          kibanaRequest: incrementKibanaRequestCounter(current.kibanaRequest, headers),
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+          overwriteEnabled: incrementBooleanCounter(current.overwriteEnabled, overwrite),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  public async incrementResolveCopySavedObjectsErrors({
+    headers,
+    createNewCopies,
+  }: IncrementResolveCopySavedObjectsErrorsOptions) {
+    const usageStats = await this.getUsageStats();
+    const { apiCalls = {} } = usageStats;
+    const { resolveCopySavedObjectsErrors: current = RESOLVE_COPY_ERRORS_DEFAULT } = apiCalls;
+
+    const attributes = {
+      ...usageStats,
+      apiCalls: {
+        ...apiCalls,
+        resolveCopySavedObjectsErrors: {
+          total: current.total + 1,
+          kibanaRequest: incrementKibanaRequestCounter(current.kibanaRequest, headers),
+          createNewCopiesEnabled: incrementBooleanCounter(
+            current.createNewCopiesEnabled,
+            createNewCopies
+          ),
+        },
+      },
+    };
+    await this.updateUsageStats(attributes);
+  }
+
+  private async updateUsageStats(attributes: UsageStats) {
+    const options = { id: SPACES_USAGE_STATS_TYPE, overwrite: true };
+    try {
+      await this.repository.create(SPACES_USAGE_STATS_TYPE, attributes, options);
```

Comment:
The default `refresh` option is `'wait_for'` which means this API call will take at up to 1s to complete. Since we don't care about read consistency we can use `wait_for=false`